pyfftw/interfaces/_utils.py

```diff
@@ -97,24 +101,25 @@ def _Xfftn(a, s, axes, overwrite_input, planner_effort,
         # take the transform of the input array! (in general, we have
         # to assume that the input array will be destroyed during 
         # planning).
-        a_copy = a.copy()
+        if overwrite_input:    # Note: reload_after_transform is not True...
+            planner_args = (a, s, axes) + args
+            a_copy = a.copy()  # ... hence this is not an additional copy
+        else:
+            planner_args = (numpy.empty_like(a), s, axes) + args
```

Comment:
(as a quick aside, you shouldn't plan on an empty array - processors often take different paths on encountering rare float types, which are less rare on random data, so the plan is essentially wrong)